azure/style.yml

```diff
@@ -1,22 +1,31 @@
 parameters:
   setup: []
+  smoketests: true
 
 jobs:
   - template: rustfmt.yml
     parameters:
       name: rustfmt
-  - template: rustfmt.yml
-    parameters:
-      name: rustfmt_beta
-      rust: beta
-      allow_fail: true
+  - ${{ if eq(parameters.smoketests, 'true') }}:
+    - template: rustfmt.yml
+      parameters:
+        name: rustfmt_beta
+        rust: beta
+        allow_fail: true
   - template: cargo-clippy.yml
     parameters:
       name: clippy
       setup: ${{ parameters.setup }}
-  - template: cargo-clippy.yml
-    parameters:
-      name: clippy_beta
-      setup: ${{ parameters.setup }}
-      rust: beta
-      allow_fail: true
+  - ${{ if eq(parameters.smoketests, 'true') }}:
```

Comment:
imo warnigns should not be hidden behind `smoketests`.  if its about build stability, clippy and warnings are probably about the same level of instability and just as important as each other

Reply:
So, this parameter was more intended to be "don't run extra things/things whose failures are ignored". A failed `cargo clippy` _should_ fail the build (imo), but a warning should not (in part because they can be caused by semver-compatbile changes to dependencies, and thus are out of your control).